nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -783,7 +784,7 @@ def _activation_trans(self, output: Tensor) -> Tensor:
         return torch.eq(self._activation(output.detach()), torch.zeros_like(output)).type_as(output)
 
     def _get_metrics_calculator(self) -> MetricsCalculator:
-        return APoZRankMetricsCalculator(dim=1)
+        return APoZRankMetricsCalculator(Scaling(kernel_size=[-1, 1], kernel_padding_mode='back'))
```

Comment:
seems every time `_get_metrics_calculator` is called, a new calculator is created. is this an expected behavior?

Reply:
seems it is an expected behavior. then it might be better to use "create" instead of "get", but anyway, not important